actions/class.Runner.php

```diff
@@ -296,8 +298,17 @@ public function init()
 
         try {
             /** @var QtiRunnerServiceContext $serviceContext */
-            $serviceContext = $this->getRunnerService()->initServiceContext($this->getServiceContext());
-            $this->returnJson($this->getInitResponse($serviceContext));
+            $serviceContext = $this->getRunnerService()->initServiceContext(
+                $this->getServiceContext()
+            );
+
+            $initResponse = $this->getInitResponse($serviceContext);
+
+            if ($this->mustForceRestoreTimersFromClient($serviceContext, $initResponse)) {
+                $initResponse['testData']['config']['timer']['restoreTimerFromClient'] = true;
```

Comment:
**issue:** Good catch! I did not read it well in the first review. Actually, the idea is to prevent the restoration from the client when the delivery is force-paused due to concurrency... The correct value for such a case would rather be `false` when a concurrency is detected. The condition is: if another delivery is executed, force-pause all existing delivery executions. Then, if the paused delivery is resumed, prevent the client from restoring the client time since it may have increased meanwhile. However, this is needed only if the delivery countdown was not frozen by the client, which would be unlikely to happen if the `preventConcurrency` plugin is active. In the case of server-side timer, this configuration would not matter that much since we don't care about client time.

Reply:
cc: @hectoras